strongbox-web-core/src/main/java/org/carlspring/strongbox/web/RepositoryMapping.java

```diff
@@ -22,4 +22,6 @@
     String storageVariableName() default "storageId";
 
     String repositoryVariableName() default "repositoryId";
+
+    boolean allowOutOfServiceRepository() default false;
```

Comment:
What exactly does this mean? I seem to be misunderstanding something... It should be possible to have a repository that is out of service. What does "allow out of service" mean?

Reply:
@carlspring There are two use cases(end points) of Repository . One is `/api/configuration/**` used to manage Repository/Storage Configuration and another one is `/storages/**` used mainly in various layouts.
We should allow to add/update/delete operation on `Out of Storage` Repository, to handle such scenario, `allowOutOfServiceRepository` is introduced.

Reply:
This annotation is used in controllers which handle `storage administration`. In those cases you don't case about the status of the repository since you are administrating it.

Reply:
It should always be possible to update/configure repositories, regardless of whether or not they are in service. Hence my asking of why we need this sort of field in he first place. I, personally, don't see the need for it. Perhaps, I'm misunderstanding your explanation...?

Reply:
You should probably check out the code. The annotation was created as part of #528 and used to convert `/{storageId}/{repositoryId}` into a `Repository` object. If the `repository` is "out of service" the argument resolver will return a `ServiceUnavailableException`. However, this behavior is also being triggered for `updating/configuring` repositories since the annotation is used there as well which is why `allowOutOfServiceRepository=true` is needed.

Reply:
Thanks for the explanation! :)